pkg/controller/runnable/reconciler.go

```diff
@@ -55,23 +58,63 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return ctrl.Result{}, err
```

Comment:
maybe unrelated, but i noticed that in https://github.com/vmware-tanzu/cartographer/pull/358 we moved towards checking for `notfound` in the repository itself rather than at this level - should we include a change for that here too?